### PR TITLE
cfme_perf.sh refactoring

### PR DESCRIPTION
**Pull request by dmetzger57**
Besides refactoring, added upgrading pip to resolve build issue with he default older pip. Tested on RHEL7 and Centos7

### CODE REVIEW

File: cfme_perf.sh [hunk 1]
```diff
@@ -7,12 +7,36 @@ else
...
+                     gcc-c++ \
+                     python-virtualenv \
+                     libffi-devel \
+                     libpng \
+                     libpng-devel \
+                     freetype-devel \
+                     freetype-devel \
+                     openssl-devel
+
+echo "### Creating Virtual Environment: .cfme_perf"
```
Comment (akrzos) [inline cfme_perf.sh:27]: We should add .cfme_perf to the .gitignore for one less item to appear in git status and prevent virtualenv files from sneaking into a commit.

**Comment by akrzos** [comment]
Worked on fedora24.  Should we squash the commits and merge it?

**Comment by dmetzger57** [comment]
just tried it on Fedora 24 Server and it needed the numpy package to succeed without error for me. Let me do a little more testing.

**Comment by akrzos** [comment]
Interesting.  numpy is in the python requirements.txt already, was there a numpy rpm you are referencing?

**Comment by dmetzger57** [comment]
I didn't add a reference, least not the I recall :-)

<SNIP>
  UPDATING build/lib.linux-x86_64-2.7/matplotlib/_version.py
  set build/lib.linux-x86_64-2.7/matplotlib/_version.py to '1.5.1'
  running build_ext
  building 'matplotlib.ft2font' extension
  setupext.py:819: UserWarning: The C headers for numpy could not be found.
You may need to install the development package
-  "The C headers for numpy could not be found. "*
  creating build/temp.linux-x86_64-2.7
  creating build/temp.linux-x86_64-2.7/src
  gcc -pthread -fno-strict-aliasing -O2 -g -pipe -Wall
  -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions
  -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches
  -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic
  -D_GNU_SOURCE -fPIC -fwrapv -DNDEBUG -O2 -g -pipe -Wall
  -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions
  -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches
  -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic
  -D_GNU_SOURCE -fPIC -fwrapv -fPIC
  -DPY_ARRAY_UNIQUE_SYMBOL=MPL_matplotlib_ft2font_ARRAY_API
  -DNPY_NO_DEPRECATED_API=NPY_1_7_API_VERSION
  -I/tmp/pip-build-_18Wpt/matplotlib/.eggs/numpy-1.11.1-py2.7-linux-x86_64.egg/numpy/core/include
  -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/local/include
  -I/usr/include -I. -I/usr/include/python2.7 -c src/ft2font.cpp -o
  build/temp.linux-x86_64-2.7/src/ft2font.o
  cc1plus: error:
  /tmp/pip-build-_18Wpt/matplotlib/.eggs/numpy-1.11.1-py2.7-linux-x86_64.egg/numpy/core/include:
  Not a directory
  error: command 'gcc' failed with exit status 1
  
  ---
  
  _Failed building wheel for matplotlib_
  <SNIP>

# 

Dennis
Why is it that all of the instruments seeking intelligent life in the
universe are pointed away from Earth?

**Comment by dmetzger57** [comment]
Removed the WIP, installs (setup) clean on Fedora Server / CentOS / RHEL

# 

Dennis
Why is it that all of the instruments seeking intelligent life in the
universe are pointed away from Earth?

File: cfme_perf.sh [hunk 2]
```diff
@@ -7,12 +7,37 @@ else
...
 virtualenv .cfme_perf
+
+echo "### Activating .cfme_per"
 echo "export PYTHONPATH='`pwd`'" | tee -a ./.cfme_perf/bin/activate
 echo "export PYTHONDONTWRITEBYTECODE=yes" | tee -a ./.cfme_perf/bin/activate
-
 . ./.cfme_perf/bin/activate
+
+echo "### Upgrading pip"
+pip install --upgrade pip
```
Comment (akrzos) [inline cfme_perf.sh:36]: For rhel we should have easy_install pip before this right?  (I assume centos would need that as well?)

**Comment by arcolife** [comment]
@akrzos looks like I didn't look at this PR before submitting https://github.com/akrzos/cfme-performance/pull/56 
err..

**Pull request merged**